### ISSUE TITLE
Shulkerboxes stacking inside hoppers after updating to 1.20.6

### ISSUE THREAD

**Issue by Patrycjusz362** [gnembon/fabric-carpet#1916]
In the latest versions of fabric-carpet stackableShulkerBoxes setting makes shulker boxes stack inside inventories without player interaction (for example a hopper pushing them into a chest). Previously that wasn't the case and the player needed to either drop them on the ground or stack them by hand.

This new behavior is a huge change which destroys some of my contraptions designed in the past. I found that the changelog for carpet 1.4.134 says that shulker boxes "should just behave like normal stackable items" which suggests that everything is working as intended but its a big pain for me after update. So can i expect a new setting which disables shulker boxes stacking inside containers or i need to change my designs to be optimized for update?

**Comment by johnnygoboy** [comment]
Speaking of which I noticed that I can't manually stack shulkers in containers now. Has this been changed or removed?

**Comment by Aculoss** [comment]
> Speaking of which I noticed that I can't manually stack shulkers in containers now. Has this been changed or removed?

Tweakeroo has a work around for now

**Comment by NickIndustries** [comment]
Breaks contraptions, please fix :)

**Comment by DaveJWalker** [comment]
I'm noticing this issue as well.  I never updated to 1.20.6, but I recently tried switching to 1.21.  This shulkerbox stacking behavior immediately causes problems.

In my particular case, I have a large auto-sorting system that distributes shulker boxes using dropper lines.  As mentioned above, the problem is that now shulkerboxes stack inside these droppers so each dropper will hold 320 shulkers instead of just 5.  The shulker distribution system begins stacking shulkerboxes and effectively 'empties' (all shulkers stack at the end of the line effectively breaking the distribution).  Refilling the distribution would take thousands of shulkerboxes.

**Comment by DaveJWalker** [comment]
One additional point... I think the original intent of the shulkerbox stacking feature was to allow them to be stacked by throwing them on the ground but not automatically within inventories like hoppers or droppers.  As mentioned, automatic stacking inside hoppers/droppers breaks a lot of mechanics.  Below is a link to Gnembon's original video:

[https://www.youtube.com/watch?v=Lt-ooRGpLz4&t=41m59s](url)

**Comment by tobymoszer** [comment]
+1 for this issue, still present in 1.21

**Comment by tigra323** [comment]
I found [this mod](https://modrinth.com/mod/stackable-shulkers-fix), that return old shulker box behavior in hoppers

**Comment by schrummy14** [comment]
@tigra323, have you tried using minecraft 1.21.1. I have not been able to get it to work properly.

**Comment by tigra323** [comment]
> @tigra323, have you tried using minecraft 1.21.1. I have not been able to get it to work properly.

No, and i don't have plans. Update fixes update suppressor, that i use
